scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,311 @@
+    so we reject the null hypothesis that they
+    have the same means. The p-value of the comparison between ``control``
+    and ``drug_a`` exceeds .05, so we accept the null hypothesis that there
+    is not a significant difference between their means.
+
+    """
+    rng = check_random_state(random_state)
+    control = np.asarray(control)
+
+    rho, df, n_group = iv_dunnett(
+        observations=observations, control=control
+    )
+
+    with suppress_warnings() as sup:
+        sup.filter(RuntimeWarning)
+        statistic = np.array([
+            stats.ttest_ind(
+                obs_, control, alternative=alternative, random_state=rng
+            ).statistic
+            for obs_ in observations
+        ])
+
+    pvalue = pvalue_dunnett(
+        rho=rho, df=df,
+        statistic=statistic, alternative=alternative,
+        rng=rng
+    )
+
+    control_mean = np.mean(control)
+    observations_mean = np.array([np.mean(obs_) for obs_ in observations])
+    std = np.sqrt((
+        np.sum(control**2)
+        + np.sum([obs_**2 for group in observations for obs_ in group])
+        - n_group*(control_mean**2 + np.sum(observations_mean**2))
+    ) / df)
+
+    return DunnettResult(
+        statistic=statistic, pvalue=pvalue,
+        _rho=rho, _df=df, _std=std,
+        _observations_mean=observations_mean,
+        _control_mean=control_mean
+    )
+
+
+def iv_dunnett(
+    observations: npt.ArrayLike, control: npt.ArrayLike
+) -> tuple[np.ndarray, int, int]:
+    """Specific parameters for Dunnett's test.
+
+    Covariance matrix depends on the number of observations in each group:
+
+    - All groups are equals (including the control), ``rho_ij=0.5`` except for
+      the diagonal which is 1.
+    - All groups but the control are equal, balanced design.
+    - Groups are not equal, unbalanced design.
+
+    Degree of freedom is the number of observations minus the number of groups
+    including the control.
+    """
+    n_n_obs = np.array([len(obs_) for obs_ in observations])
+
+    # From Dunnett1955 p. 1100 d.f. = (sum N)-(p+1)
+    n_obs = n_n_obs.sum()
+    n_control = len(control)
+    n = n_obs + n_control
+    n_groups = len(observations)
+    df = n - n_groups - 1
+
+    # rho_ij = 1/sqrt((N0/Ni+1)(N0/Nj+1))
+    rho = n_control/n_n_obs + 1
+    rho = 1/np.sqrt(rho[:, None] * rho[None, :])
+    np.fill_diagonal(rho, 1)
+
+    return rho, df, n_groups
+
+
+def pvalue_dunnett(
+    rho: np.ndarray, df: int, statistic: np.ndarray,
+    alternative: Literal['two-sided', 'less', 'greater'],
+    rng: SeedType = None
+) -> np.ndarray:
+    """pvalue from Dunnett critical value.
+
+    Critical values come from the multivariate student-t distribution.
+    """
+    statistic = statistic.reshape(-1, 1)
+
+    mvt = stats.multivariate_t(shape=rho, df=df, seed=rng)
+    if alternative == "two-sided":
+        pvalue = 1 - mvt.cdf(statistic, lower_limit=-statistic)
```

Comment:
Sometimes `statistic` is negative, which makes the integral negative, and the `pvalue` is greater than 1. This is correct behavior for a general integral, but maybe we should adjust `multivariate_t.cdf` to take the `abs`?

Reply:
Should we then just take the `abs` of `statistic` as returned by the `ttest`?

Reply:
I think I fixed that part. It was indeed needed and saw that when testing against `ttest_ind`.